third_party/move/move-compiler-v2/legacy-move-compiler/src/parser/syntax.rs

```diff
@@ -3590,7 +3590,10 @@ fn parse_module(
                 },
                 // Regular move constructs
                 Tok::Use => ModuleMember::Use(parse_use_decl(attributes, context)?),
-                Tok::Friend if context.tokens.lookahead()? != Tok::Fun => {
+                Tok::Friend
+                    if context.tokens.lookahead()? != Tok::Fun
+                        && context.tokens.lookahead()? != Tok::Inline =>
```

Comment:
We currently allow `public(friend)` versions of the below, so we should also admit these (but we do not, and fail with syntax error):

```move
friend native fun bar();

friend entry fun whatever();
```

Please also add tests for the above (native, entry) cases.

Reply:
done